contracts/pcv/PCVDepositAggregator.sol

```diff
@@ -0,0 +1,365 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "./IPCVDepositAggregator.sol";
+import "./PCVDeposit.sol";
+import "../refs/CoreRef.sol";
+import "../external/Decimal.sol";
+import "./balancer/IRewardsAssetManager.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableMap.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract PCVDepositAggregator is IPCVDepositAggregator, CoreRef {
+    using EnumerableSet for EnumerableSet.AddressSet;
+    using SafeERC20 for IERC20;
+
+    // ---------- Events ----------
+
+    // ---------- Properties ------
+
+    struct PCVDepositInfo {
+        uint128 weight;
+        bool depositsPaused;
+    }
+
+    EnumerableSet.AddressSet private pcvDepositAddresses;
+    
+    mapping(address => PCVDepositInfo) public pcvDepositInfos;
+    
+    uint128 public bufferWeight;
+    uint128 public totalWeight; 
+    address public token;
+    address public override rewardsAssetManager;
+
+    constructor(
+        address _core,
+        address _rewardsAssetManager,
+        address _token,
+        address[] memory _initialPCVDepositAddresses,
+        uint128[] memory _initialPCVDepositWeights,
+        uint128 _bufferWeight
+    ) CoreRef(_core) {
+        if (_initialPCVDepositAddresses.length != _initialPCVDepositWeights.length) {
+            revert("Addresses and weights are not the same length!");
+        }
+
+        if (_rewardsAssetManager == address(0x0)) {
+            revert("Rewards asset manager cannot be null");
+        }
+
+        rewardsAssetManager = _rewardsAssetManager;
+        token = _token;
+        bufferWeight = _bufferWeight;
+        totalWeight = bufferWeight;
+
+        _setContractAdminRole(keccak256("PCV_CONTROLLER_ROLE"));
+
+        for (uint i=0; i < _initialPCVDepositAddresses.length; i++) {
+            _addPCVDeposit(_initialPCVDepositAddresses[i], _initialPCVDepositWeights[i]);
+        }
+    }
+
+    // ---------- Public Functions -------------
+
+    function rebalance() external virtual override {
+        _rebalance();
+    }
+
+    function rebalanceSingle(address pcvDeposit) external virtual override {
+        _rebalanceSingle(pcvDeposit);
+    }
+
+    function deposit() external virtual override {
```

Comment:
This shouldn't be a no-op right? It should rebalance if above buffer or something

Reply:
Nevermind